kdump-lib-initramfs.sh

```diff
@@ -102,8 +102,20 @@ get_fs_type_from_target()
 
 get_mntpoint_from_target()
 {
-	# --source is applied to ensure non-bind mount is returned
-	get_mount_info TARGET source "$1" -f
+	local _source _target
+	# omit sources that are bind mounts i.e. they contain a [/path/to/subpath].
+	findmnt -k --pairs -o SOURCE,TARGET "$1" | while read -r line; do
+		_source="${line#SOURCE=\"}"
+		_source="${_source%%\" TARGET=*}"
+
+		_target="${line#*TARGET=\"}"
+		_target="${_target%%\"}"
+
```

Comment:
Hi Lichen,
a few nits here.
1.  shellcheck complains about the use of `local` as it's not POSIX compliant
2. you can read multiple variables with `read`, i.e. you can drop `line`
3. when you drop `--pairs` `findmnt` should omit the `SOURCE=` and `TARGET=` prefix as well as the quotation (please double check)

putting it all together this should simplify the code to a single line
```
findmnt -k -o SOURCE,TARGET "$1" | while read -r _source _target; do
```
Other than that the series looks fine

Reply:
Oups, missed the comment from @cgwalters above.  He's right, that 3. would break if either `_source` or `_target` contains a space. So probably better to keep option `--pairs`.

Reply:
Hi @prudo1 ,thanks for reviewing. I removed the `local _source _target` and use `read -r _source _target` now.
I keep `--pairs` becasue in my test, I need to handle cases when `_source` or `_target` contains a space.